### DISCUSSION TITLE
Lesson 5: ValueError: Unable to expand environment variable in host setting: 'https://rinkeby.infura.io/v3/$WEB3_INFURA_PROJECT_ID'

### DISCUSSION THREAD

**Discussion by BrysonHall** [smartcontractkit/full-blockchain-solidity-course-py#1199] [Q&A] [answered]
I was doing the Patrick Collins 16 hours tutorial [link](https://youtu.be/M576WGiDBdQ?t=17887) and I got this error when running brownie run scripts/deploy.py --network rinkeby:

 "[ValueError: Unable to expand environment variable in host setting: 'https://rinkeby.infura.io/v3/$WEB3_INFURA_PROJECT_ID']"

I have been looking at a lot of threads about this but could not solve it. I`m on windows. Help is much apricated 🙂

My code:

deploy.py

      from brownie import accounts, config, SimpleStorage, network

       def deploy_simple_storage():
       account = get_account()
      simple_storage = SimpleStorage.deploy({"from": account})
      stored_value = simple_storage.retrieve()
      print(stored_value)
      transaction = simple_storage.store(15, {"from": account})
      transaction.wait(1)
      updated_stored_value = simple_storage.retrieve()
      print(updated_stored_value)

    def get_account():
     if network.show_active() == "development":
        return accounts[0]
    else:
        return accounts.add(config["wallets"]["from_key"])

     def main():
     deploy_simple_storage()

.env

        export WEB3_INFURA_PROJECT_ID='xxxxxxxxxxxxxxxxxxxxxxx`
        export` PRIVATE_KEY='0xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'

brownie-config.yaml

    dotenv: .env
    wallets:
    from_key: ${PRIVATE_KEY}

**Comment by n4n0b1t3** [comment, marked as answer]
Things that might cause the problem:

1. your env file is formatted incorrectly, there are two backticks in there and you don't need single or any quotes to surround the addresses, and why the indentation?
2. What is exactly your command line when running the script? 
example: `brownie run .\scripts\deploy.py --network rinkeby`
3. you seem to use rinkeby but have not set it up in your brownie config, however, that might not be strictly necessary. How does the ganache-cli line look like in the terminal response?
4. All the formatting looks odd to me, the indentations are all over the place. Is there any reason for this? Your python file should look like this.

Just for reference here are the code fragments you are using with correct formatting:
The .env file:
```rd
export WEB3_INFURA_PROJECT_ID=n4n0b1t3n4n0b1t3n4n0b1t3n4n0b1t3
export WEB3_ALCHEMY_PROJECT_ID=n4n0b1t3n4n0b1t3n4n0b1t3n4n0b1t3

export PRIVATE_KEY=0xn4n0b1t3n4n0b1t3n4n0b1t3n4n0b1t3n4n0b1t3n4n0b1t3n4n0b1t3n4n0b1t3

### If you want to verify contracts on etherscan
export ETHERSCAN_TOKEN=N4N0B1T3N4N0B1T3N4N0B1T3N4N0B1T3
```
The config.yaml file
```yaml
dotenv: .env
wallets:
  from_key: ${PRIVATE_KEY}
```


```python
from brownie import accounts, config, SimpleStorage, network

def deploy_simple_storage():
    account = get_account()
    simple_storage = SimpleStorage.deploy({"from": account})
    stored_value = simple_storage.retrieve()
    print(stored_value)
    transaction = simple_storage.store(15, {"from": account})
    transaction.wait(1)
    updated_stored_value = simple_storage.retrieve()
    print(updated_stored_value)

def get_account():
    if network.show_active() == "development":
        return accounts[0]
    else:
        return accounts.add(config["wallets"]["from_key"])

def main():
    deploy_simple_storage()
```

**Reply by n4n0b1t3** [reply]
and here is the network response:
```rd
(py39brownie) fund_me>brownie run .\scripts\tmp_test.py --network rinkeby
INFORMATION: Es konnten keine Dateien mit dem angegebenen
Muster gefunden werden.
Brownie v1.18.1 - Python development framework for Ethereum

FundMeProject is the active project.

Running 'scripts\tmp_test.py::main'...
Transaction sent: 0x04ed147672b86e22d5236b5ee93b97944688ab8926692b477b0c289be2df7a94
  Gas price: 1.000000011 gwei   Gas limit: 368944   Nonce: 60
  SimpleStorage.constructor confirmed   Block: 10276481   Gas used: 335404 (90.91%)
  SimpleStorage deployed at: 0xeFAfD8b9Fb1Bfa273A11EBB1a41645Eae81859E1

0
Transaction sent: 0x042071db81879ce0fc93e4c99a1f25fe4874616b5638a82e389d483c22596299
  Gas price: 1.000000011 gwei   Gas limit: 47867   Nonce: 61
  SimpleStorage.store confirmed   Block: 10276482   Gas used: 43516 (90.91%)

  SimpleStorage.store confirmed   Block: 10276482   Gas used: 43516 (90.91%)

15
(py39brownie) fund_me>
```

**Reply by BrysonHall** [reply]
Hey dude! This solved my problem. Much appreciated.  Have a nice weekend :)

**Reply by n4n0b1t3** [reply]
OK, great! It looks like your question has been answered. It helps others who have a similar problem immensely if they can see at a glance whether a question has already been answered. That is why it is important that answers which have contributed to the solution are marked as such. Please use the button "Mark as Answer".

**Reply by cromewar** [reply]
You are just awesome @n4n0b1t3 This is a very detailed explanation.
